dask_planner/src/sql/logical/sort.rs

```diff
@@ -10,64 +12,61 @@ pub struct PySort {
-        match &expr.expr {
-            Expr::Sort {
-                expr,
-                asc,
-                nulls_first,
-            } => {
-                println!("expr {}", expr);
-                println!("asc {}", asc);
-                println!("nulls_first {}", nulls_first);
-                match expr.as_ref() {
-                    Expr::Column(col) => {
-                        println!("Is a column {}", col);
-                        println!("col_name {}", col.name);
-                        col.name.clone()
-                    }
-                    _ => panic!("Sort Expression expr not of column type!"),
-                }
-            }
-            _ => panic!("Provided expression is not a sort expression"),
-        }
-    }
-    #[pyo3(name = "isAscending")]
-    pub fn is_ascending(&self, expr: PyExpr) -> bool {
-        match &expr.expr {
+    /// Returns if a sort expressions denotes an ascending sort
+    fn is_ascending(&self, expr: Expr) -> bool {
+        match expr {
             Expr::Sort {
                 expr: _,
                 asc,
                 nulls_first: _,
-            } => asc.clone(),
-            _ => panic!("Provided expression is not a sort expression"),
+            } => asc,
+            _ => panic!("Provided expression is not a sort epxression"),
         }
     }
-    #[pyo3(name = "isNullsFirst")]
-    pub fn is_nulls_first(&self, expr: PyExpr) -> bool {
-        match &expr.expr {
+    /// Returns if nulls should be placed first in a sort expression
+    fn is_nulls_first(&self, expr: Expr) -> bool {
+        match &expr {
             Expr::Sort {
                 expr: _,
                 asc: _,
                 nulls_first,
             } => nulls_first.clone(),
-            _ => panic!("Provided expression is not a sort expression"),
+            _ => panic!("Provided expression is not a sort epxression"),
         }
     }
 }
+#[pymethods]
+impl PySort {
+    /// Returns a Vec of the sort expressions
+    #[pyo3(name = "getCollation")]
+    pub fn sort_expressions(&self) -> PyResult<Vec<PyExpr>> {
+        let mut sort_exprs: Vec<PyExpr> = Vec::new();
+        for expr in &self.sort.expr {
+            sort_exprs.push(PyExpr::from(expr.clone(), Some(self.sort.input.clone())));
+        }
+        Ok(sort_exprs)
+    }
+
+    #[pyo3(name = "getAscending")]
+    pub fn get_ascending(&self) -> PyResult<Vec<bool>> {
+        let mut is_ascending: Vec<bool> = Vec::new();
+        for sortexpr in &self.sort.expr {
+            is_ascending.push(self.is_ascending(sortexpr.clone()))
+        }
+        Ok(is_ascending)
+    }
+    #[pyo3(name = "getNullsFirst")]
+    pub fn get_nulls_first(&self) -> PyResult<Vec<bool>> {
+        let nulls_first: Vec<bool> = self
+            .sort
+            .expr
+            .iter()
+            .map(|sortexpr| self.is_nulls_first(sortexpr.clone()))
+            .collect::<Vec<bool>>();
+        Ok(nulls_first)
```

Comment:
The original code is just fine but we can simplify it slightly since Rust can infer the types here without hints.

```suggestion
        Ok(self
            .sort
            .expr
            .iter()
            .map(|sortexpr| self.is_nulls_first(sortexpr.clone()))
            .collect())
```